src/GenSymIO.chpl

```diff
@@ -646,14 +647,33 @@ module GenSymIO {
         return reply;
     }
 
+    /**
+     * inline proc to validate the range for our domain.
+     * Valid domains must be increasing with the lower bound <= upper bound
+     * :arg r: 1D domain
+     * :type domain(1): one dimensional domain
+     *
+     * :returns: bool True iff the lower bound is less than or equal to upper bound
+     */
+    inline proc _isValidRange(r: domain(1)): bool {
```

Comment:
Note that this is the same as `!r.isEmpty()` if you wanted to just use a builtin. In terms of terminology we don't tend to think of this as valid vs malformed, but just empty vs non-empty.